### DISCUSSION TITLE
how to install sriov operator on talos

### DISCUSSION THREAD

**Discussion by yingwana3** [siderolabs/talos#7552] [Q&A]
Hi

Our application need vf resources which should be bind to userspace driver such as vfio-pci. But we don't have permission to echo vfs on talos. RHEL OCP provide sriov operator to require such needs. Native k8s also has sriov operator (https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/master/doc/quickstart.md ) but it needs to install first.
As talos can be only interacted by talosctl, we don't know how to install sriov operator on it.

**Comment by smira** [comment]
What is the specific problem you're facing?

**Comment by steverfrancis** [comment]
Did you try to deploy the sriov operator into Kubernetes and it failed?

**Reply by yingwana3** [reply]
hi
![image](https://github.com/siderolabs/talos/assets/60091241/411d8669-8b47-46e0-957d-4a18daccaa0b)
We don't know how to execute go or make commands on talos.

**Reply by smira** [reply]
In this guide, they are not executed on the cluster, they are executed on a machine which has access to the cluster via `kubeconfig` (Kubernetes client config).

**Reply by yumin-l** [reply]
I installed sriov operator on talos. But it seems that the operator did not work due to "read-only file system". I also tried to manually create vfs by "echo 2 > sriov_numfs" and it also failed due to "read-only file system". Do you have some tips to create vfs on the system?
![image](https://github.com/siderolabs/talos/assets/101164877/3145dbf8-2bd1-4d2b-8cc4-1d3788e4d0a8)

**Reply by smira** [reply]
Talos rootfs is read-only, so udev rules can't be injected this way, but they can be added via system extensions https://github.com/siderolabs/extensions
